assessments/projects/opa/self-assessment.md

```diff
@@ -0,0 +1,563 @@
+still loading; however, a user would typically set this up with their
+own policies.
+
+* By default, OPA does not restrict access to any of its REST API endpoints
+that are used to fetch, create and update policy and data. It’s possible that
+an attacker can corrupt the policy and data loaded into OPA, thereby bypassing
+OPA’s authorization checks altogether
+
+* If OPA is deployed in an insecure environment, its effectiveness can be
+compromised by an attacker
+
+* OPA can be configured to fetch policies and data from remote HTTP servers
+using its *bundle* feature. The files inside the bundle are tar.gz compressed.
+An attacker who has access to the remote server or MitM the connection between
+OPA and the remote server.  This can cause a Denial of Service (DoS) by
+providing a bundle file that will consume the server’s memory and therefore
+crash OPA or be used to add, remove, or delete OPA policies
+
+* OPA trusts the authenticity of the policies and data it is provided with.
+An attacker can potentially feed corrupt policies and data to OPA
+
+* Policies that rely on existing status of cluster resources can be raced due
+to the design of the validating webhook and eventual consistency model.
+Attackers can potentially bypass these policies by crafting a series of
+requests. Suggestions to warn against such policies - or a warning to require
+routine checking during runtime may be required
+
+* If policies for different components interact, the user must author the
+policies in such a way that there are not problematic interactions
+
+The following issues raised during the assessment have been addressed:
+
+* OPA decision logs may contain sensitive information like usernames, passwords
+provided in the input to the policy. OPA now allows the user to mask sensitive
+fields in the input from showing up in the decision logs by allowing the user
+to define a policy that specifies the fields to mask
+
+* To make it easier for users to get familiar with Rego and also to make the
+policy authoring experience convenient and less error prone, we
+created [The Rego Playground](https://play.openpolicyagent.org/) and more
+recently we’ve updated the [OPA documentation](https://www.openpolicyagent.org/docs/latest/policy-language/)
+to make the policy examples interactive !
+
+### Compensating Mechanisms
+
+* The OPA daemon can be configured to authenticate and authorize requests. This
+could prove beneficial especially in scenarios where the deployment environment
+is untrusted. More details on deploying OPA securely can be found [here](https://www.openpolicyagent.org/docs/latest/security/). Additionally OAuth2.0 client credentials can be used to authenticate
+OPA’s REST APIs. See [this](https://github.com/open-policy-agent/opa/issues/1205)
+issue.
+
+* Since OPA does not contain any policies by default when started, it’s
+recommended that the application fail-open in case of any non-200 response
```

Comment:
Fail open? Surely it should fail closed?

Reply:
Corrected